hooks/before_prepare/icon.js

```diff
@@ -124,18 +124,26 @@ function generate() {
 		source = path.join(__dirname, '../../res/icon.svg');
 		fn = 'density';
 	}
-
-	return Promise.all(platform.icons.forEach(function (icon) {
-		var dest = path.join(root, icon.file);
-
-		if (!fs.existsSync(path.dirname(dest))) {
-			mkdirp.sync(path.dirname(dest));
-		}
-
-		var image = gm(source)[fn](icon.dimension, icon.dimension);
-
-		return pify(image.write.bind(image), Promise)(dest);
-	})).then(function () {
+	
+	var promises = function () {
+		var p = [];
+		
+		platform.icons.forEach(function (icon) {
+			var dest = path.join(root, icon.file);
+
+			if (!fs.existsSync(path.dirname(dest))) {
+				mkdirp.sync(path.dirname(dest));
+			}
+
+			var image = gm(source)[fn](icon.dimension, icon.dimension);
+
+			p.push(pify(image.write.bind(image), Promise)(dest));
+		});
+		
+		return p;
+	}();
```

Comment:
Or better, just keep the old code and use `.map` instead of `.forEach`. Easier to see what changed in the diff then.